src/Uno.UI/UI/Xaml/Controls/Image/Image.cs

```diff
@@ -296,105 +330,72 @@ protected override Size MeasureOverride(Size availableSize)
+					return desiredSize;
+				}
+				if (isHeightDefined)
 				{
-					var knownWidth = ImageControl.GetKnownWidth(availableSize.Width, sourceSize.Width);
-					var knownHeight = ImageControl.GetKnownHeight(availableSize.Height, sourceSize.Height);
-					var knownSize = new Size(knownWidth, knownHeight);
-
-					switch (ImageControl.Stretch)
+					var knownHeight = img.GetKnownHeight(constrainedAvailableSize.Height, sourceSize.Height);
+					var desiredSize = new Size();
+					switch (img.Stretch)
 					{
 						case Stretch.Uniform:
-							var desiredSize = new Size();
-							var aspectRatio = sourceSize.AspectRatio();
+							//If sourceSize is empty, aspect ratio is undefined so we return 0
 							// Since apsect ratio can have a lot of decimal, iOS ceils Image size to 0.5 if it's not a precise size (like 111.111111111)
 							// so the desiredSize will never match the actual size causing an infinite measuring and can freeze the app
-							desiredSize.Width = Math.Min(knownWidth, Math.Ceiling(knownHeight * aspectRatio * 2) / 2);
-							desiredSize.Height = Math.Min(knownHeight, Math.Ceiling(knownWidth / aspectRatio * 2) / 2);
-
-							if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
-							{
-								this.Log().Debug(Panel.ToString() + $" measuring with knownSize={knownSize} and Stretch.Uniform with availableSize={availableSize}, returning desiredSize={desiredSize}");
-							}
+							desiredSize.Width = sourceSize == default(Size) ? 0 : Math.Ceiling(knownHeight * aspectRatio * 2) / 2;
+							desiredSize.Height = knownHeight;
+							break;
+						case Stretch.None:
+							desiredSize.Width = sourceSize.Width;
+							desiredSize.Height = sourceSize.Height;
+							break;
+						case Stretch.Fill:
+						case Stretch.UniformToFill:
+							desiredSize.Width = double.IsInfinity(constrainedAvailableSize.Width) ? sourceSize.Width : constrainedAvailableSize.Width;
+							desiredSize.Height = knownHeight;
+							break;
+					}
 
-							return desiredSize;
-							//TODO: #11103 - For futur needs, for an Image with no specified size, need to implement Stretching: UniformToFill, Fill, None
-							//case Stretch.UniformToFill:
-							//case Stretch.Fill:
-							//case Stretch.None:
-							//return sourceSize;
+					if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+					{
+						this.Log().Debug(Panel.ToString() + $" measuring with knownHeight={knownHeight} with availableSize={constrainedAvailableSize}, returning desiredSize={desiredSize}");
 					}
-				}
 
-				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
-				{
-					this.Log().Debug(Panel.ToString() + $" measuring with availableSize={availableSize}, returning sourceSize={sourceSize}");
+					return desiredSize;
 				}
 
-				return sourceSize;
+				throw new InvalidOperationException("Should never reach here.");
```

Comment:
Why ?

Reply:
It's not logically possible to reach there, but the compiler can't figure it out.